packaging/greenboot/functions.sh

```diff
@@ -294,3 +294,94 @@ function log_script_exit() {
         echo "FINISHED"
     fi
 }
+
+# lvmsDriverShouldExist shallowly wraps some python yaml parsing to determine if the LVMS-Operator is expected to be
+# deployed.
+# args:
+#   $1 microshift config file path
+# return:
+#   0:  "true", expect the driver to be running
+#   1:  "false", do not expect the driver to be running
+function lvmsDriverShouldExist() {
+    # The python script checks the config's .storage.driver value. Parsing errors are ignored, these will be caught
+    # by MicroShift during startup, before this greenboot script is run. The script favors verbosity over simplified
+    # code.
+    # args:
+    #   None
+    # return:
+    #   None
+    # exit code:
+    #   0: The lvms components should be checked for. This case occurs for any one of:
+    #       - .storage.driver == one of ["", "lvms"]
+    #       - .storage.driver is unset (indicates default config)
+    #       - .storage is unset (indicates default config)
+    #       - the file cannot be opened (indicates default config)
+    #   1: The lvms components should not be checked for. This case occurs for:
+    #       - .storage.driver == "none"
+    if ! python -c \
+'import yaml, sys
```

Comment:
Should we explicitly require `python3-pyyaml` in the .spec file for the microshift-greenboot if we're going to use it?
I know there's pyyaml pulled by something else, but not sure what it is.

Reply:
Good catch, yeah I'll add it to the greenboot package reqs.